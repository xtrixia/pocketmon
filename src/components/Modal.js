import { css } from "@emotion/react";

import Typography from "./Typography";

import { COLORS } from "../root/colors";
import { SPACINGS } from "../root/spacings";
import { BREAKPOINTS } from "../root/breakpoints";

const stadium = css`
  border-radius: 2rem;
  background: ${COLORS.primary};
  padding: ${SPACINGS.lg};
  margin: ${SPACINGS.md};
  @media (min-width: ${BREAKPOINTS.sm}) {
    width: 500px;
    align-self: center;
  }
`;

function Modal({ content, children }) {
  return (
    <div
      className={css`
        justify-content: center;
        position: fixed;
        background: rgba(0, 0, 0, 0.75);
        width: 100vw;
        height: 100vh;
        z-index: 1000;
        left: 0;
        top: 0;
        display: flex;
        flex-direction: column;
      `}
    >
      <div className={stadium}>
        <Typography variant="h2">{content}</Typography>
        {children}
      </div>
    </div>
  );
}
export default Modal;
