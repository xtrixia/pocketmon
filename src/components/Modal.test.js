import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";

import Modal from "./Modal";

test("Modal should render correctly", () => {
  const component = renderer.create(
    <Router>
      <Modal content="This is modal" />
    </Router>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree.children[0].children[0].children[0]).toEqual("This is modal");
});

test("Modal should render transparently", () => {
  const component = renderer.create(
    <Router>
      <Modal transparent>
        <p>This is modal</p>
      </Modal>
    </Router>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree.children[0].children[0].type).toEqual("p");
  expect(tree.children[0].children[0].children[0]).toEqual("This is modal");
});
