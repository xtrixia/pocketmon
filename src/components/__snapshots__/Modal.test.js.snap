// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render correctly 1`] = `
<div
  className="css-16e2dkp"
>
  <div
    className="css-153gz4b"
  >
    <p
      className="css-1alkoj4"
    >
      This is modal
    </p>
  </div>
</div>
`;

exports[`Modal should render transparently 1`] = `
<div
  className="css-16e2dkp"
>
  <div
    className="css-kdxjg9"
  >
    <p>
      This is modal
    </p>
  </div>
</div>
`;
